lang/java/reef-common/src/main/java/org/apache/reef/driver/restart/DriverRestartManagerImpl.java

```diff
@@ -45,30 +45,39 @@
 
   private boolean restartBegan;
   private boolean restartCompleted;
+  private boolean isRestartSet;
+  private boolean isRestart;
 
   @Inject
   private DriverRestartManagerImpl(final DriverRuntimeRestartManager driverRuntimeRestartManager) {
     this.driverRuntimeRestartManager = driverRuntimeRestartManager;
     this.restartCompleted = false;
     this.restartBegan = false;
+    this.isRestart = false;
+    this.isRestartSet = false;
     this.previousEvaluators = new HashSet<>();
     this.recoveredEvaluators = new HashSet<>();
   }
 
   @Override
-  public boolean isRestart() {
-    return driverRuntimeRestartManager.isRestart();
+  public synchronized boolean isRestart() {
```

Comment:
It's just adding a layer of caching on top so it doesn't need to invoke the runtime restart manager every time.